### PR TITLE
Support `line` layer on Globe with Terrain3D

### PR DESCRIPTION
**Pull request by birkskyum**
### Context for this series of PRs

Towards
- #4927

For the Globe with Terrain3D, I have been working on getting all the layers working.

These changes generally touches the `draw_[layer].ts`, and in rare cases the `[layer]_program.ts` too.

The fixes for these layers have each their own complexities. Luckily they can be reviewed and applied individually, which I think is the better way to go about it.

The `line` is the simplest, so I've tried to prepare a PR here with the bare minimum of changes needed to get it going.

### Support **`line`** layer on Globe with Terrain3D
I set the ignoreGlobeMatrix flag in this case, to avoid applying the globeMatrix twice, since it's also applied to the Terrain3D.

<details>
<summary>Repro example using Liberty style from OpenFreeMap</summary>

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <title>3D Terrain</title>
    <meta property="og:description" content="Go beyond hillshade and show elevation in actual 3D." />
    <meta charset='utf-8'>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel='stylesheet' href='../../dist/maplibre-gl.css' />
    <script src='../../dist/maplibre-gl-dev.js'></script>
    <style>
        body { margin: 0; padding: 0; }
        html, body, #map { height: 100%; }
    </style>
</head>
<body>
<div id="map"></div>
<script>
    const map = (window.map = new maplibregl.Map({
        container: 'map',
        zoom: 12,
        center: [11.39085, 47.27574],
        pitch: 70,
        hash: true,
        
        maxZoom: 18,
        maxPitch: 85
    }));

    map.setStyle('https://tiles.openfreemap.org/styles/liberty', {
            transformStyle: (previousStyle, nextStyle) => {
                
                nextStyle.sources = {
                    ...nextStyle.sources, terrainSource: {
                        type: 'raster-dem',
                        url: 'https://demotiles.maplibre.org/terrain-tiles/tiles.json',
                        tileSize: 256
                    },
                    hillshadeSource: {
                        type: 'raster-dem',
                        url: 'https://demotiles.maplibre.org/terrain-tiles/tiles.json',
                        tileSize: 256
                    }
                }
                nextStyle.terrain = {
                    source: 'terrainSource',
                    exaggeration: 1
                }

                // nextStyle.sky = {}

                // nextStyle.layers.push({
                //     id: 'hills',
                //     type: 'hillshade',
                //     source: 'hillshadeSource',
                //     layout: { visibility: 'visible' },
                //     paint: { 'hillshade-shadow-color': '#473B24' }
                // })

                return nextStyle
            }
        })

    map.addControl(
        new maplibregl.NavigationControl({
            visualizePitch: true,
            showZoom: true,
            showCompass: true
        })
    );

    map.addControl(
        new maplibregl.GlobeControl()
    );

    map.addControl(
        new maplibregl.TerrainControl({
            source: 'terrainSource',
            exaggeration: 1
        })
    );
</script>
</body>
</html>
```

</details>

Before
Screenshot 2024-11-03 at 18 34 22

After
Screenshot 2024-11-03 at 18 26 58

## Launch Checklist

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
Do you think we should create a branch in this repo to allow merging everything to that branch like we did with the globe feature?

**Comment by birkskyum** [comment]
I actually don't think it's necessary. Unlike with the globe, it appear that it's quite possible to land the improvements in an incremental manner - independently of each other.

The layers are quite well separated in the code.

**Comment by HarelM** [comment]
What about adding a render test? Even a single geojson and zero elevation tiles should do, right?

**Comment by birkskyum** [comment]
@HarelM I have added 3 render tests. They do render the line correctly.

They show up a bit odd for now, because the fill and background layers are missing still - but the lines are correctly rendered.

**Comment by HarelM** [comment]
I think @louwers is using either the globe or terrain folders to ignore parity for gender tests, can you use globe/terrain/... Or the other way around?

**Comment by birkskyum** [comment]
sure, they actually fit nicely in besides the existing test of the `raster` layer

File: src/render/draw_line.ts [hunk 1]
```diff
@@ -26,7 +26,9 @@ export function drawLine(painter: Painter, sourceCache: SourceCache, layer: Line
 
     const depthMode = painter.getDepthModeForSublayer(0, DepthMode.ReadOnly);
     const colorMode = painter.colorModeForRenderPass();
-
+    const hasTerrain = !!painter.style.map.terrain;
```
Comment (HarelM) [inline src/render/draw_line.ts:29]: Is there a place where these two are used separately? If not, consider using a single boolean.

**Review by HarelM — APPROVED** [review]
